tests/integration/backup/test_v3_backup.py

```diff
@@ -181,7 +190,10 @@ def test_roundtrip_from_kafka_state(
         check=True,
     )
 
-    # Verify restored topic.
+    # Verify configuration is identical.
+    assert topic_config == get_topic_configurations(admin_client, new_topic.name, {ConfigSource.TOPIC_CONFIG})
```

Comment:
I moved this so that it doesn't happen in-between reading and verifying of records.